cmd/restic/global.go

```diff
@@ -366,26 +368,38 @@ func OpenRepository(opts GlobalOptions) (*repository.Repository, error) {
 
 	s := repository.New(be)
 
-	passwordTriesLeft := 1
-	if stdinIsTerminal() && opts.password == "" {
-		passwordTriesLeft = 3
-	}
+	if opts.MasterKeyFile == "" {
 
-	for ; passwordTriesLeft > 0; passwordTriesLeft-- {
-		opts.password, err = ReadPassword(opts, "enter password for repository: ")
-		if err != nil && passwordTriesLeft > 1 {
-			opts.password = ""
-			fmt.Printf("%s. Try again\n", err)
-		}
-		if err != nil {
-			continue
+		passwordTriesLeft := 1
+		if stdinIsTerminal() && opts.password == "" {
+			passwordTriesLeft = 3
 		}
 
-		err = s.SearchKey(opts.ctx, opts.password, maxKeys, opts.KeyHint)
-		if err != nil && passwordTriesLeft > 1 {
-			opts.password = ""
-			fmt.Printf("%s. Try again\n", err)
+		for ; passwordTriesLeft > 0; passwordTriesLeft-- {
+			opts.password, err = ReadPassword(opts, "enter password for repository: ")
+			if err != nil && passwordTriesLeft > 1 {
+				opts.password = ""
+				fmt.Printf("%s. Try again\n", err)
+			}
+			if err != nil {
+				continue
+			}
+
+			err = s.SearchKey(opts.ctx, opts.password, maxKeys, opts.KeyHint)
+			if err == nil {
+				if !opts.JSON {
+					Verbosef("password is correct\n")
+				}
+				break
+			}
+
+			if err != nil && passwordTriesLeft > 1 {
+				opts.password = ""
+				fmt.Printf("%s. Try again\n", err)
+			}
 		}
```

Comment:
this whole block is rather long, please consider extracting it into a separate function or method.

Reply:
Thanks for the suggestion! I also changed it.